gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageOptions.java

```diff
@@ -34,12 +35,47 @@
   private static final String DEFAULT_PATH_DELIMITER = "/";
 
   private final String pathDelimiter;
+  private final TimeSource timeSource;
   private transient StorageRpc storageRpc;
 
+  /**
+   * A class providing access to the current time in milliseconds.
+   *
+   * Implementations should implement {@code Serializable} wherever possible and must document
```

Comment:
s/should/must/ ? otherwise would be a problem for ServiceOptions.

Also, I think we should have it defined in ServiceOptions and also document that this is mostly use for testing.

If inner class, Shall we call it Clock? and mentioned that this will be removed in favor Java 8 Clock in the future.